cmd/api-response.go

```diff
@@ -596,25 +598,28 @@ func writeErrorResponseHeadersOnly(w http.ResponseWriter, err APIError) {
 
 // writeErrorResponseJSON - writes error response in JSON format;
 // useful for admin APIs.
-func writeErrorResponseJSON(w http.ResponseWriter, err APIError, reqURL *url.URL) {
+func writeErrorResponseJSON(ctx context.Context, w http.ResponseWriter, err APIError, reqURL *url.URL) {
 	// Generate error response.
-	errorResponse := getAPIErrorResponse(err, reqURL.Path, w.Header().Get(responseRequestIDKey))
+	errorResponse := getAPIErrorResponse(ctx, err, reqURL.Path, w.Header().Get(responseRequestIDKey), w.Header().Get(responseDeploymentIDKey))
 	encodedErrorResponse := encodeResponseJSON(errorResponse)
 	writeResponse(w, err.HTTPStatusCode, encodedErrorResponse, mimeJSON)
 }
 
 // writeCustomErrorResponseJSON - similar to writeErrorResponseJSON,
 // but accepts the error message directly (this allows messages to be
 // dynamically generated.)
-func writeCustomErrorResponseJSON(w http.ResponseWriter, err APIError,
+func writeCustomErrorResponseJSON(ctx context.Context, w http.ResponseWriter, err APIError,
 	errBody string, reqURL *url.URL) {
 
+	reqInfo := logger.GetReqInfo(ctx)
 	errorResponse := APIErrorResponse{
-		Code:      err.Code,
-		Message:   errBody,
-		Resource:  reqURL.Path,
-		RequestID: w.Header().Get(responseRequestIDKey),
-		HostID:    "3L137",
+		Code:       err.Code,
+		Message:    errBody,
+		Resource:   reqURL.Path,
+		BucketName: reqInfo.BucketName,
```

Comment:
`reqInfo` can be `nil`

Reply:
We need to set `BucketName`  and `ObjectName` only if it is not `nil`. do we not have this info already as part of `Resource`?

Reply:
Really when ? Isn't newContext support to always add reqInfo . If it's nil then it's a bug.

Reply:
We are also using `context.Background()`

Reply:
Ah then yes..

Reply:
I looked at `GetReqInfo` code. It returns, empty `ReqInfo` in that case, so we should be good.

Reply:
`reqInfo` will only be `nil` if `ctx` is `nil`